src/pandablocks_ioc/ioc.py

```diff
@@ -222,18 +223,19 @@ def _store_values(
             "LABEL_"
         ):
             _, block_name_number = field_name.split("_", maxsplit=1)
-            if block_name_number in block_info:
-                number_of_blocks = block_info[block_name_number].number
-            else:
-                number_of_blocks = block_info[block_name_number[:-1]].number
 
-            # The block is fixed with metadata
+            # The block is fixed with metadata, it should end with a number
             #     "*METADATA.LABEL_SEQ2": "NewSeqMetadataLabel",
             if not block_name_number[-1].isdigit():
                 raise ValueError(
                     f"Recieved metadata for a block name {block_name_number} that "
                     "didn't contain a number"
                 )
+
+            parts = re.findall(r"\d+|[^\d]+", block_name_number)
```

Comment:
I also debugged the issue. The original code worked well for blocks with less than 10 instances. The following code would expand it to work with blocks with 2 digit numbers. The block names may contain numbers in the middle, for example `SFP2_SYNC_OUT`.
```
block_name_no_number = block_name_number if block_name_number in block_info else re.sub(r"\d*$", "", block_name_number)
```

Reply:
Yep good point! I'll make the change in dev before merging to main.